### PR TITLE
feat(ui): DOMA-8189 added tip component

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
изображение
изображение

### CODE REVIEW

File: packages/ui/src/components/Tip/style.less [hunk 1]
```diff
@@ -0,0 +1,53 @@
+@import "antd/lib/tooltip/style/index.less";
+@import (reference) "@open-condo/ui/src/tokens/variables.less";
+
+.condo-tooltip.condo-tip {
+  max-width: 350px;
+  padding: 4.3px;
```
Comment (Alllex202) [inline packages/ui/src/components/Tip/style.less:6]: 4.3? :)

File: packages/ui/src/components/Tip/style.less [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      font-family: @condo-global-font-family-default;
+      line-height: @condo-global-typography-text-s-line-height;
+      background-color: @condo-global-color-black;
+      border-radius: @condo-global-border-radius-large;
+      box-shadow: @condo-global-box-shadow-default;
+
+      & .condo-tip-icon-wrapper {
+        position: absolute;
+        top: 8px;
+        right: 8px;
```
Comment (Alllex202) [inline packages/ui/src/components/Tip/style.less:23]: How about this?
```suggestion
        top: @condo-global-spacing-8;
        right: @condo-global-spacing-8;
```

File: packages/ui/src/components/Tip/tip.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    useEffect(() => {
+        setOpen(initialOpen)
+    }, [initialOpen])
+
+    const handleClose = useCallback(() => {
+        if (onClose) {
+            onClose()
+        }
+
+        setOpen(prev => !prev)
```
Comment (Alllex202) [inline packages/ui/src/components/Tip/tip.tsx:39]:
```suggestion
        setOpen(false)
```

File: packages/ui/src/components/Tip/style.less (hunk 2 above)
Comment (sitozzz) [inline packages/ui/src/components/Tip/style.less:23]: It's better to use global spacing variables

File: packages/ui/src/components/Tip/style.less (hunk 1 above)
Comment (nomerdvadcatpyat) [reply packages/ui/src/components/Tip/style.less]: changed to 4

**Review by sitozzz — APPROVED** [review]

**Review by Alllex202 — APPROVED** [review]

**Review by SavelevMatthew — CHANGES_REQUESTED** [review]
What's the difference from current Tooltip component?
(except for the headline and visuals)

**Comment by nomerdvadcatpyat** [comment]
> What's the difference from current Tooltip component? (except for the headline and visuals)

Different opening/closing behavior.
Tooltip is opened and closed by some trigger (in our case it is hover), this component closes either by clicking on the cross icon or by performing an external action (prop open). Combined with the different styles and the addition of tite/message (not just the title) there are a lot of changes to make it a Tooltip component new type.
In addition to this in UI-kit designs it is made as a separate component

**Comment by toplenboren** [comment]
image

I think we should add a little padding here, so that tip triangle does not collide with button

**Review by toplenboren — APPROVED** [review]

File: packages/ui/src/components/Tour/tourContext.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React, { useContext, useState } from 'react'
+
+type TourContextValue = {
+    currentStep?: number
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Tour/tourContext.tsx:4]: Why everything is optional?

Comment (SavelevMatthew) [reply packages/ui/src/components/Tour/tourContext.tsx:4]: You should provide proper default value instead

Comment (SavelevMatthew) [reply packages/ui/src/components/Tour/tourContext.tsx:4]: Context if basically a 3 things (types + defaults + actual values)

Comment (SavelevMatthew) [reply packages/ui/src/components/Tour/tourContext.tsx:4]: So you can solve it like this:

1. Types. You should say value is defined
```typescript
type TourContextType = {
    currentStep: number
    setCurrentStep: React.Dispatch<React.SetStateAction<number>>
}
```
2. Defaults. What to return if no Provider not found in parent components
```typescript
const TourContext = React.createContext<TourContextValue>({
    currentStep: 0
    setCurrentStep: () => ({})
})
```
So, now if you decided to use context without a provider, you will get this defaults

3. Actual values
```
const TourContextProvider = ...
```

**Review by SavelevMatthew — APPROVED** [review]
LGTM

**Pull request merged**